libs/ledger-live-common/src/families/stellar/specs.ts

```diff
@@ -56,12 +77,147 @@ const stellar: AppSpec<Transaction> = {
         };
       },
       test: ({ account, accountBeforeTransaction, operation, transaction }) => {
-        expect(account.balance.toString()).toBe(
-          accountBeforeTransaction.balance.minus(operation.value).toString()
+        // We don't know what the final fee will be until after the tx is
+        // submitted. Using higher max fee to make sure tx doesn't time out.
+        expect(account.balance.toNumber()).toBeLessThanOrEqual(
+          accountBeforeTransaction.balance.minus(operation.value).toNumber()
+        );
+
+        if (transaction.memoValue) {
+          expect(operation.extra).toMatchObject({
+            memo: transaction.memoValue,
+          });
+        }
+
+        // Transaction type "payment" can be either "payment" or "create_account"
+        // (if account is not created).
+        const getType = () => {
+          switch (transaction.operationType) {
+            case "payment":
+              return /payment|create_account/;
+            case "changeTrust":
+              return /change_trust/;
+            default:
+              return "";
+          }
+        };
+
+        expect(transaction.operationType).toMatch(getType());
+      },
+    },
+    {
+      name: "add USDC asset",
```

Comment:
Add asset test is flaky. Sometimes it's failing when executed right after a passing test (no changes at all). It looks like the bot is canceling the transaction. Any idea why? Maybe it's caused by the environment?

![image](https://user-images.githubusercontent.com/7346473/170780636-fee3e30d-7626-4474-a338-a5c16b77d63f.png)
_Transaction in the bot (everything is correct)_

![image](https://user-images.githubusercontent.com/7346473/170780725-cc6435b7-bfcc-4a52-b1d9-758f806c6d32.png)
_Output of the bot_

Reply:
Hello @quietbits,
after looking into the hw-app-stellar, it's to fall into an unknown case where it's doesnt find any good status

`libs/ledgerjs/packages/hw-app-str/src/Str.ts`
`line:254`

I don't know if we actually support this, but it may require some change to make it work.
Does it work outside the bot ?

Reply:
Hello @henrily-ledger ! Thank you for looking into this. Everything works fine in the Ledger Live app, the issue is only when testing the bot.